internal/pkg/nats/options.go

```diff
@@ -0,0 +1,132 @@
+//
+// Copyright (c) 2022 One Track Consulting
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//go:build include_nats_messaging
+
+package nats
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/edgexfoundry/go-mod-messaging/v2/internal/pkg"
+	"github.com/edgexfoundry/go-mod-messaging/v2/pkg/types"
+	"github.com/nats-io/nats.go"
+)
+
+// ClientConfig contains all the configurations for the MQTT client.
+type ClientConfig struct {
+	BrokerURL string
+	ClientOptions
+}
+
+// ConnectionOptions contains the connection configurations for the MQTT client.
+//
+// NOTE: The connection properties resides in its own struct in order to avoid the property being loaded in via
+//  reflection during the load process.
+type ConnectionOptions struct {
+	BrokerURL string
+}
+
+// ClientOptions contains the client options which are loaded via reflection
+type ClientOptions struct {
+	// Client Identifiers
+	Username             string
+	Password             string
+	ClientId             string
+	Format               string
+	RetryOnFailedConnect bool
+	Durable              string
+	AutoProvision        bool
+	ConnectTimeout       int // Seconds
+	pkg.TlsConfigurationOptions
+	QueueGroup string
+}
+
+// CreateClientConfiguration constructs a ClientConfig based on the provided MessageBusConfig.
+func CreateClientConfiguration(messageBusConfig types.MessageBusConfig) (ClientConfig, error) {
+	var brokerUrl string
+	if !messageBusConfig.PublishHost.IsHostInfoEmpty() {
+		brokerUrl = messageBusConfig.PublishHost.GetHostURL()
+	} else if !messageBusConfig.SubscribeHost.IsHostInfoEmpty() {
+		brokerUrl = messageBusConfig.SubscribeHost.GetHostURL()
+	} else {
+		return ClientConfig{}, fmt.Errorf("Specified empty broker info.")
+	}
+
+	_, err := url.Parse(brokerUrl)
+	if err != nil {
+		return ClientConfig{}, pkg.NewBrokerURLErr(fmt.Sprintf("Failed to parse broker: %v", err))
+	}
+
+	clientOptions := CreateClientOptionsWithDefaults()
+	err = pkg.Load(messageBusConfig.Optional, &clientOptions)
+	if err != nil {
+		return ClientConfig{}, err
+	}
+
+	tlsConfig := pkg.TlsConfigurationOptions{}
+	err = pkg.Load(messageBusConfig.Optional, &tlsConfig)
+	if err != nil {
+		return ClientConfig{}, err
+	}
+
+	clientOptions.TlsConfigurationOptions = tlsConfig
+
+	return ClientConfig{
+		BrokerURL:     brokerUrl,
+		ClientOptions: clientOptions,
+	}, nil
+}
+
+func (cc ClientConfig) ConnectOpt() []nats.Option {
+	connectTimeout := time.Second
```

Comment:
I'm taking this change with the others for now but going to look into the default option as well, theres a lot of benefits to aligning this better to the MQTT implementation I think.  If I remeber right you need to have it set to something with NATS.go to avoid it being zero but could be wrong or an issue thats since been fixed (I typically use 1s for local and 5s for NGS connections)